### ISSUE TITLE
Make sure initial deploy with insufficient permissions fails nicely

### ISSUE THREAD

**Issue by tj** [apex/up#319]
Right now you'll get `Error: cannot find the API, looks like you haven't deployed` on subsequent attempts to interact with the deploy, and can use `up stack` to see the stack's status, but it's not obvious that permission (or something else) went wrong.

**Comment by larron** [comment]
Not sure if this is even possible... but would also be cool to have `up` detect insufficient power? An example scenario would be the failure that occurs when setting memory to something to low to serve the app.

**Comment by tj** [comment]
@larron hmm yea that specific case might be tricky

**Comment by larron** [comment]
Yup probably so.. I had a case where my node app was being crunched and I continued to see relay errors.  After upping the power the reverse proxy was finally able to connect.  Was another gotcha

**Comment by tj** [comment]
@larron noted! Might make more sense to increase the default and let people downsize, that would at least highlight the issue more clearly

**Comment by larron** [comment]
Excellent! :)

**Comment by kaihendry** [comment]
hi, I currently switch between staging & production accounts with `ln -sf up.dev.json up.json` and `ln -sf up.prod.json up.json`. prod works fine, but in dev I've hit this:

         Error: cannot find the API, looks like you haven't deployed

Log: https://static.spuul.com/devops/2017-11-21/wth.txt

What am I missing?

**Comment by tj** [comment]
Hmm I forget what this issue was. @kaihendry currently that message is only from your up.json `name` field's API Gateway missing, that's what it checks to determine if you've deployed previously. Maybe you renamed one but didn't deploy?

This error sucks though, could be more helpful or at least point to some docs.

**Comment by kaihendry** [comment]
Nevermind, I think I worked around it with a `up stack delete` and rummaging around Cloudformation. :sweat_smile:
